MSRE multiphysics RZ model - Refs #311 (#312)
idaholab/virtual_test_bed: doc/content/msr/msre/multiphysics_rz_model/msre_multiphysics_core_model.md

# Molten Salt Reactor Experiment (MSRE) Multiphysics Model

*Contact: Mauricio Tano, mauricio.tanoretamales\@inl.gov*

*Model summarized and documented by Andres Fierro, Dr. Samuel Walker, and Dr. Mauricio Tano*

*Model link: [Griffin-Pronghorn Steady-State Model](https://github.com/idaholab/virtual_test_bed/tree/devel/msr/msre/multiphysics_core_model/steady_state)*

!tag name=MSRE Griffin-Pronghorn Steady State Model pairs=reactor_type:MSR
                       reactor:MSRE
                       geometry:core

>                        simulation_type:core_multiphysics

Reviewer: ```suggestion
                       simulation_type:multiphysics
```

>                        input_features:multiapps

Reviewer: ```suggestion
                       input_features:multiapps
                       transient:steady_state
```

>                        code_used:BlueCrab

Reviewer: ```suggestion
                       codes_used:BlueCrab;Griffin;Pronghorn
```

                       computing_needs:Workstation
                       fiscal_year:2023

This model of the MSRE utilizes MOOSE to create a 2D, RZ (cylindrical coordinates) mesh of the MSRE.
Griffin computes neutronics and resulting normalized power source [!citep](Javi23).

> Pronghorn perform medium-fidelity, coarse mesh thermal-hydraulics analysis of the core, upper plenum, pump, downcomer, and lower plenum [!citep](mau23).

Reviewer: ```suggestion
Pronghorn performs medium-fidelity, coarse mesh thermal-hydraulics analysis of the core, upper plenum, pump, downcomer, and lower plenum [!citep](mau23).
```

Author: Incorporated change on following commit.

Griffin and Pronghorn are coupled via the `MultiApp` system.
The parts of the MSRE loop are represented in [MSRE_pgh_mesh_blocks].
The model is an axisymmetric model with the left vertical axis being the axis of symmetry.

> The fuel salt flows down the `Downcomer`, through the `Lower Plenum`, and up into the `Core`; then, it is collected at the `Upper Plenum`, and, finally, it is recycled through the `Pump`.

Reviewer: ```suggestion
The fuel salt flows down the `Downcomer`, through the `Lower Plenum`, and up into the `Core`; then, it is collected at the `Upper Plenum`, and, finally, it is circulated through the `Pump` into the loop again.
```

Author: Incorporated change on next PR.

A porous medium approach is used to model flow conditions in the core with a vertical porosity of `0.22283`.
No rugosity is assumed when computing the friction factors.

> An anisotropic friction source coefficient keeps the flow approximately 1-Dimensional at the core.

Reviewer: ```suggestion
An anisotropic friction source coefficient keeps the flow approximately 1-Dimensional at the core, to reproduce the fluid behaviors in the core channels.
```

Author: Incorporated change on following commit.

!media msr/msre/MSRE_pgh_mesh_blocks.png
       style=width:80%;margin-left:auto;margin-right:auto
       id=MSRE_pgh_mesh_blocks
       caption=Subdomain (left) and mesh (right) of the multiphysics model.

> 

Reviewer: Please insert a paragraph here about creating the mesh using the mesh.i file
Jun asked for it there https://github.com/idaholab/virtual_test_bed/issues/311#issuecomment-1875969316

Reviewer: done

## Neutronics Model

Griffin is the main tool used in the neutronics input file listed below.
Griffin calls the Pronghorn thermal-hydraulics sub-app which is detailed in the thermal hydraulics section that is discussed later.

!listing msr/msre/multiphysics_core_model/steady_state/neu.i

#### Problem Parameters

The beginning of the input file lists the problem parameters the user can edit including the model's physical properties and initial conditions.

#### Global Parameters

Next, Global Parameters are defined in the `Global Parameters` block which give the macroscopic cross section file path for the MSRE.

!listing msr/msre/multiphysics_core_model/steady_state/neu.i block=GlobalParams

#### Transport Systems

Next, the `Transport System` block determines what sort of neutronic solve Griffin will perform.
Here the methodology to compute the eigenvalue of the MSRE system is computed and corresponding boundary conditions are set --- reflecting boundary for the RZ symmetric core and vacuum boundaries elsewhere.

Additionally, the type of transport equation solve is selected.
In this case the a multigroup Diffusion equation is sufficient for fast multiphysics coupling.

> Additionally, options for the jacobian and fission source auxiliary variable are selected here.

Reviewer: ```suggestion
Additionally, options for the Jacobian and fission source auxiliary variable are selected here.
```

Author: Incorporated change on following commit.

$\Sigma_{rg}$ removal cross-section from energy group $g$, $k_{eff}$ eigenvalue representing the effective multiplication factor of the reactor, $\chi_{p,g}$ prompt fission spectrum for neutrons born in energy group $g$, $\nu$ number of neutrons per fission, $\Sigma_{fg'}$ average fission cross-section of neutrons in energy group $g'$, $\Sigma_{sg'}$ differential scattering cross-section for neutrons scattering from energy group $g'$ to energy group $g$, $\beta_0$ delayed neutrons fraction, $\chi_{d,g}$ delayed fission spectrum for neutrons born in energy group $g$ due to decay of neutron precursors,  $\lambda_i$ decay constant for precursor group $i$, $c_i(\mathbf{r})$ concentration of neutron precursor group $i$, at position $\mathbf{r}$, and $G$ total number of energy groups.

#### Mesh

The `Mesh` block can either generate a mesh using the MOOSE meshing capabiility or reads in a previously generated mesh. Here, the Mesh block reads in a MOOSE generated mesh and specifies that the model is RZ.

!listing msr/msre/multiphysics_core_model/steady_state/neu.i block=Mesh

#### Aux Variables

> The `AuxVariables` block is specificed next. Through the ```Auxiliary Variables```, developers can define external variables that are solved or used in the primary simulation. The variable types, functions, and scaling factors are explained in detail [here](https://mooseframework.inl.gov/syntax/AuxVariables/index.html). Aux variables can be set explicitly in this block, or passed from other apps to perform multiphysics coupling.

Reviewer: ```suggestion
The `AuxVariables` block is specificed next. Through the ```Auxiliary Variables```, developers can define external variables that are solved or used in the primary simulation. The variable types, computation, and scaling factors are explained in detail [here](https://mooseframework.inl.gov/syntax/AuxVariables/index.html). Auxiliary variables can be set explicitly in the `AuxKernel` section, or set from another app using Transfers to perform multiphysics coupling.
```

Author: Incorporated change on following commit.

Finally, the `MultiApps` block sets up any sub apps that will be used in a multiphysics coupling. Here we just have one multi app which is the Pronghorn solve of the thermal hydraulics and delayed neutron precursor group distributions.

!listing msr/msre/multiphysics_core_model/steady_state/neu.i block=MultiApps

#### Transfers

Correspondingly, the `Transfers` block determines how the main and sub apps will communicate with each other. These transfers send auxiliary variables to and from different apps. Here the `power_density` and `fission_source` calculated by Griffin is sent to the sub-app Pronghorn to inform the thermal hydraulic solution.

In return, Pronghorn sends the delayed neutron precursor group distributions `c1-c6`, the solid and liquid temperatures, and the velocities to Griffin to determine the steady state multiphysics solution of the reactor.

!listing msr/msre/multiphysics_core_model/steady_state/neu.i block=Transfers

#### Outputs

Lastly, the `Outputs` block specifies what type of output (e.g. exodus and CSV) will be created during the simulation.

!listing msr/msre/multiphysics_core_model/steady_state/neu.i block=Outputs

## Thermal Hydraulics Model

>  Next, Pronghorn (AKA Moose Navier Stokes Module) is the sub app detailed in the neutronics input file listed below which performs the thermal hydraulic calculations of the core and primary loop.

Reviewer: ```suggestion
 Next, Pronghorn (here mostly leveraging the MOOSE Navier Stokes Module) is the sub app detailed in the neutronics input file listed below which performs the thermal hydraulic calculations of the core and primary loop.
```

Author: Incorporated change on following commit.

>  Note, this application can be fully run with the open-source Moose Navier Stokes Module.

Reviewer: ```suggestion
 Note, this application can be fully run with the open-source MOOSE Navier Stokes Module.
```

Author: Incorporated change on following commit.

!listing msr/msre/multiphysics_core_model/steady_state/th.i start=[Mesh] end=[Problem]

#### Modules

This block defines the Module that will be used in this problem - specifically the Navier Stokes Finite Volume action.

The ```Navier Stokes Finite Volume``` action is used to define fluid properties, compute the "weakly-compressible" flow, and set up boundary conditions.

Here, the action operates on the fluid blocks which are defined at the top of the header and are substituted using the `$` sign. Recall that the `$` sign refers to variable substitutions. "Weakly-compressible" with porous_medium_treatment is selected for this model and the energy equation is also included.

Numerical schemes can be selected for solving the Navier Stokes equations with different methods of pressure and velocity interpolation.

Porosity, friction, and turbulence treatment can be modified here using various correlations and models. Fluid properties are also set here and can be modified.

The action also incorporates external physics like a volumetric heat source due to the power_density calculated by Griffin. Also, boundary conditions are set to constrain the different Navier Stokes equations.

> The scalar equations for delayed neutron precursor groups are not included here in the action since they are defined externally elsewhere in the input file

Reviewer: yes but why

Author: This was done explicitly so that advection to be modified to account for precursors jump due to porosity.
